### PR TITLE
Droth 609 road node information to kalpa api

### PR DESCRIPTION
**Pull request by leirasf**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 1]
```diff
@@ -525,7 +533,7 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
 
   def getViiteRoadLinksHistoryFromVVH(roadAddressesLinkIds: Set[Long]): Seq[VVHHistoryRoadLink] = {
     if (roadAddressesLinkIds.nonEmpty) {
-      val historyData = Await.result(vvhClient.historyData.fetchVVHRoadlinkHistoryF(roadAddressesLinkIds), atMost = Duration.Inf)
+      val historyData = Await.result(vvhClient.historyData.fetchVVHRoadLinkHistoryByLinkIdsF(roadAddressesLinkIds), atMost = Duration.Inf)
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: Should this be named historyData.fetchByLinkIdsF?

Comment (tvenhola) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: :+1:

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 2]
```diff
@@ -195,7 +216,8 @@ class VVHClient(vvhRestApiEndPoint: String) {
     filter
   }
 
-  private def withLimitFilter(attributeName: String, low: Int, high: Int, includeAllPublicRoads: Boolean = false): String = {
+  //TODO check this method to removoe the adminclass
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Should this comment be checked before merge?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 3]
```diff
@@ -271,250 +310,190 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
+    * Creates a pseudo VVH Timestamp for new assets and speed limits. Turns clock back to 0:00 on the same day
+    * if less than offsetHours have passed since or 0:00 on previous day if not.
+    *
+    * @param offsetHours Number of hours since midnight to return current day as a VVH timestamp (UNIX time in ms)
+    */
+  def createVVHTimeStamp(offsetHours: Int): Long = {
+    VVHClient.createVVHTimeStamp(offsetHours)
+  }
+
+  //TODO change this comment
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Should this TODO be checked?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 4]
```diff
@@ -271,250 +310,190 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
+    val definition = layerDefinition(withMunicipalityFilter(Set(municipality)))
+    val url = serviceUrl(definition, queryParameters())
 
     fetchVVHFeatures(url) match {
       case Left(features) => features.map(extractVVHFeature)
       case Right(error) => throw new VVHClientException(error.toString)
     }
   }
 
+  //TODO change this comments
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Should this TODO be checked?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 5]
```diff
@@ -525,77 +504,54 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
 
     fetchVVHFeatures(url) match {
       case Left(features) => features.map(extractVVHFeature)
       case Right(error) => throw new VVHClientException(error.toString)
     }
   }
 
+  protected def queryLinksIdByPolygons(polygon: Polygon): Seq[Long] = {
+    val polygonString = stringifyPolygonGeometry(polygon)
+    //TODO have a look on this validation
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Should this TODO be checked?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 6]
```diff
@@ -525,77 +504,54 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
-    * Returns VVH road links.
-    * Used by RoadLinkService.fetchVVHRoadlinks (called from CsvGenerator)
-    */
-  def fetchVVHRoadlinks[T](linkIds: Set[Long],
-                           fieldSelection: Option[String],
-                           fetchGeometry: Boolean,
-                           resultTransition: (Map[String, Any], List[List[Double]]) => T): Seq[T] =
-    fetchVVHRoadlinks(linkIds, fieldSelection, fetchGeometry, resultTransition, withLinkIdFilter)
-
+  //TODO comment and generic type !??!?!??!?
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Should this TODO be checked?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala [hunk 7]
```diff
@@ -161,7 +161,8 @@ class OracleLinearAssetDao(val vvhClient: VVHClient) {
         where a.asset_type_id = $assetTypeId and a.id = $id
         """.as[(Long, Double, Double)].list
 
-    val roadLinksByLinkId = vvhClient.fetchByLinkIds(links.map(_._1).toSet)
+    //TODO This sould be done in DAO object
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala]: Should this TODO be checked?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala [hunk 8]
```diff
@@ -619,7 +620,8 @@ class OracleLinearAssetDao(val vvhClient: VVHClient) {
         where a.asset_type_id = 20 and a.id = $id
         """.as[(Long, Long, SideCode, Option[Int], Double, Double, Option[String], Option[DateTime], Option[String], Option[DateTime], Long, Option[DateTime])].list
 
-    val roadLinksByLinkId = vvhClient.fetchByLinkIds(speedLimits.map(_._2).toSet)
+    //TODO This sould be done in DAO object
```
Comment (halied) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala]: Should this TODO be checked?

**Comment by halied** [comment]
Some comments added, otherwise ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala (hunk 8 above)
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala]: This is something we have found when we were developing but didn't want to touch it now. In future developments if we have to do something here we can move it away from the DAO.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala (hunk 7 above)
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala]: This is something we have found when we were developing but didn't want to touch it now. In future developments if we have to do something here we can move it away from the DAO.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 9]
```diff
@@ -1023,111 +973,34 @@ class VVHHistoryClient(vvhRestApiEndPoint: String) extends VVHClient(vvhRestApiE
...
-        ""
-      } else {
-        val query = ids.mkString(",")
-        s"where=$attributeName IN ($query)"
-      }
-    filter
-  }
-
   private def linkIdFilter(linkIds: Set[Long]): String = {
     withFilter("LINKID", linkIds)
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Check if it is not repeated

**Comment by joanacoelho** [comment]
ok

**Pull request merged**